### PR TITLE
usb: Add remote wakeup support

### PR DESCRIPTION
**Pull request by pawelzadrozniak**
This commit adds support for remote wakeup and extends USB api
with a wakeup request call. Remote wakeup can be dsabled in kconfig
when a specific driver does not support this feature.

### CODE REVIEW

**Review by masz-nordic — APPROVED** [review]

File: subsys/usb/usb_device.c [hunk 1]
```diff
@@ -1368,6 +1387,18 @@ int usb_transfer_sync(u8_t ep, u8_t *data, size_t dlen, unsigned int flags)
 	return pdata.tsize;
 }
 
+int usb_wakeup_request(void)
```
Comment (jfischer-phytec-iot) [inline subsys/usb/usb_device.c]: Who calls it, class driver, user app?

**Review by jfischer-no — CHANGES_REQUESTED** [review]
Please split it in two commits, with changes in device stack and changes in the driver. It would be nice to have a sample.

File: subsys/usb/usb_device.c (hunk 1 above)
Comment (pawelzadrozniak) [reply subsys/usb/usb_device.c]: User application.

Comment (pawelzadrozniak) [reply subsys/usb/usb_device.c]: Split into 2 commits; extended hid_mouse example.

**Review by dbkinder — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Review by finikorg — APPROVED** [review]
LGTM

**Comment by carlescufi** [comment]
Adding dev-review so that we can consider merging this for 1.14

**Review by carlescufi — APPROVED** [review]

**Comment by pdunaj** [comment]
Hi @carlescufi , at the moment we have big problems with USB as states handling is broken since January. I though this change that also reworks state handling will come quickly but for some reason it is stuck.
I know release is coming but at the moment USB is broken and without fix it would not be usable really without hacks on the application side...

**Pull request merged**